x-pack/docs/en/rest-api/security/oidc-api.asciidoc

```diff
@@ -0,0 +1,233 @@
+[role="xpack"]
+[[security-api-oidc]]
+=== OpenID Connect APIs
+
+{es} exposes all the necessary OpenID Connect related functionality using the following APIs. These APIs
+are used internally by {kib} in order to provide OpenID Connect based authentication, but can also be used by other,
+custom web applications or other clients. If you are looking for the configuration of the OpenID Connect Realm
+see <placeholder>
+
+==== Authorization
+
+To use any of the following APIs, you must have at least the `manage_oidc` cluster privilege.
+
+[[security-api-oidc-prepare]]
+==== Prepare Authentication API
+
+Creates an oAuth 2.0 authentication request as a URL string based on the configuration of the respective
+OpenID Connect authentication realm in {es}
+
+===== Request
+
+`POST /_security/oidc/prepare`
+
+
+===== Request Body
+
+The following parameters can be specified in the body of the request:
+
+`realm`::
+The name of the OpenID Connect realm in {es} the configuration of which should be used in order to
+generate the authentication request. Cannot be specified when `iss` is specified.
+
+`state`::
+String value used to maintain state between the authentication request and the response, typically used
+as a Cross-Site Request Forgery mitigation. If the caller of the API doesn't provide a value, {es} will
+generate one with sufficient entropy itself and return it in the response.
+
+`nonce`::
+String value used to associate a Client session with an ID Token, and to mitigate replay attacks.
+If the caller of the API doesn't provide a value, {es} will generate one with sufficient entropy itself
+and return it in the response.
+
+`issuer`::
+In the case of a 3rd Party initiated Single Sign On, this is the Issuer Identifier for the OP that the RP is
+to send the Authentication Request to. Cannot be specified when `realm` is specified.
+
+`login_hint`::
+In the case of a 3rd Party initiated Single Sign On, a string value to be included in the authentication
+request, as the `login_hint` parameter. This parameter is not valid when `realm` is specified
+
+
+===== Examples
```

Comment:
We'd need to decide if we want to run tests on these or not. I lean to saying yes, so I'll add an OIDC realm configuration in the testInteg cluster in a commit

Reply:
I changed my mind on this one. I see the same challenges in the SAML Realm APIs that I 'm working on. It doesn't make much sense to test the APIs here as these feel like forced Integration tests rather than API doc tests. Thoughts @albertzaharovits , @lcawl  ?